src/client.rs

```diff
@@ -50,119 +46,73 @@ pub struct Collection {
+    verifier: Box<dyn Verification>
 }
 
 impl Default for Client {
-
     fn default() -> Self {
         Client {
-            server_url: "https://firefox.settings.services.mozilla.com/v1".to_string(),
-            bucket_id: "main".to_string(),
-            collection_name: "".to_string()
+            server_url: "https://firefox.settings.services.mozilla.com/v1".to_owned(),
+            bucket_name: "main".to_owned(),
+            collection_name: "".to_owned(),
+            verifier: Box::new(DefaultVerifier{})
         }
     }
-    
-}
-
-#[async_trait]
-impl Verification for Client {
-    async fn verify(collection: &Collection) -> Result<(), SignatureError> {
-
-        println!("In Verfication!");
-        // Fetch certificate PEM (public key).
-        let x5u = collection.metadata["signature"]["x5u"].as_str().unwrap();
-        let resp = reqwest::get(&x5u.to_string()).await?;
-        let pem = resp.bytes().await?;
-
-        // Parse PEM (OpenSSL)
-        let cert = X509::from_pem(&pem)?;
-        let public_key = cert.public_key()?;
-        let ec_public_key = public_key.ec_key()?;
-        let mut ctx = BigNumContext::new()?;
-        let group = EcGroup::from_curve_name(Nid::SECP384R1)?;
-        let public_key_bytes = ec_public_key.public_key().to_bytes(
-            &group,
-            PointConversionForm::UNCOMPRESSED,
-            &mut ctx,
-        )?;
-        let pk: EcdsaPublicKey<NistP384> = EcdsaPublicKey::from_bytes(&public_key_bytes)?;
-
-        // Instantiate signature
-        let b64_signature = collection.metadata["signature"]["signature"]
-            .as_str()
-            .unwrap_or("");
-        let signature_bytes = base64::decode_config(&b64_signature, base64::URL_SAFE)?;
-        let signature = FixedSignature::<NistP384>::from_bytes(&signature_bytes)?;
-
-        // Serialized data.
-        let mut sorted_records = collection.records.to_vec();
-        sorted_records.sort_by(|a, b| (a["id"]).to_string().cmp(&b["id"].to_string()));
-        let serialized = serialize(&json!({
-            "data": sorted_records,
-            "last_modified": collection.timestamp.to_string().to_owned()
-        }));
-        let data = format!("Content-Signature:\x00{}", serialized);
-
-        // Verify
-        verify_sha384(&P384Verifier::from(&pk), &data.as_bytes(), &signature)?;
-
-        Ok(())
-    }
 }
 
 impl Client {
 
-    pub async fn create_with_collection(collection_name: &str) -> Self {
-        println!("The collection name is {}", collection_name);
+    fn instantiate_verifier(verifier: Option<Box<dyn Verification>>) -> Box<dyn Verification> {
+        return match verifier {
+            Some(verifier) => verifier,
+            None => Box::new(DefaultVerifier{})
+        }
+    }
 
-        let client = Client {
-            collection_name: collection_name.to_string(),
+    pub fn create_with_collection(collection_name: &str, verifier: Option<Box<dyn Verification>>) -> Self {
```

Comment:
no, since we expect any struct that implements Verification trait and so we don't know the size of the struct at compile time - Box<dyn Verification> is needed